### PR TITLE
LLT-5466 Fix malformed utf-8 byte sequence in logs

### PR DESCRIPTION
**Pull request by jjanowsk**
Because of the lack of lock in logger implementation two logs could
overlap leading to missing logs or in our case to malformed utf-8
sequences.

### :ballot_box_with_check: Definition of Done checklist
- [x] Commit history is clean ([requirements](../blob/main/docs/git_commit_messages_requirements.md))
- [x] README.md is updated
- [x] Functionality is covered by unit or integration tests

### CODE REVIEW

**Review by Jauler — APPROVED** [review]
nice catch! +1.0

File: nat-lab/tests/uniffi/libtelio_remote.py [hunk 1]
```diff
@@ -75,19 +76,21 @@ def next_event(self):
...
-                    f"{datetime.datetime.now()} Failed to write logline due to error {str(e)}\n"
-                )
+        with self.lock:
+            with open(TCLI_LOG, "a", encoding="utf-8") as logfile:
+                try:
+                    logfile.write(f"{datetime.datetime.now()} {log_level} {payload}\n")
+                except IOError as e:
+                    logfile.write(
+                        f"{datetime.datetime.now()} Failed to write logline due to error {str(e)}\n"
+                    )
```
Comment (tomaszklak) [inline nat-lab/tests/uniffi/libtelio_remote.py:93]: nit: Maybe we should switch to [logging](https://docs.python.org/3/library/logging.html#thread-safety) which claims to be thread safe by design?

**Review by tomaszklak — APPROVED** [review]
+1

**Comment by tomaszklak** [comment]
One additional question - i would expect that we only produce ascii logs which should be fine regardless of how many threads write them to the same file.

Is that not in fact the case, and we produce non ascii utf8 code points? 😮

**Comment by jjanowsk** [comment]
> One additional question - i would expect that we only produce ascii logs which should be fine regardless of how many threads write them to the same file.
> 
> Is that not in fact the case, and we produce non ascii (utf-8? utf-16?) code points? 😮

@tomaszklak We do. WireGuard Go logs this character: "…" which is utf-8 encoded as: 0xE2 0x80 0xa6. It's possible that this is the only place and that's why only WireGuard Go tests have failed.

File: nat-lab/tests/uniffi/libtelio_remote.py (hunk 1 above)
Comment (jjanowsk) [reply nat-lab/tests/uniffi/libtelio_remote.py:93]: I don't see big wins here from using it.

**Pull request merged**